src/domain/flow-system/services/src/flow/flow_service_impl.rs

```diff
@@ -44,6 +48,7 @@ pub struct FlowServiceImpl {
     task_scheduler: Arc<dyn TaskScheduler>,
     flow_configuration_service: Arc<dyn FlowConfigurationService>,
     dataset_changes_service: Arc<dyn DatasetChangesService>,
+    dataset_repo: Arc<dyn DatasetRepository>,
```

Comment:
I'm very much against this dependency, if we can avoid it.

Reply:
Yeah, it will be removed and used in reset service level